Tests/test_SffIO.py

```diff
@@ -368,181 +368,104 @@ def test_index2(self):
 
 
 class TestSelf(unittest.TestCase):
-    """
-    These tests were originally seltests run in SffIO.py
-    """
+    """ These tests were originally seltests run in SffIO.py """
 
     def test_read(self):
         filename = "Roche/E3MFGYR02_random_10_reads.sff"
-        with open(filename, "rb") as handle:
-            sff = list(SffIterator(handle))
-        with open(filename, "rb") as handle:
-            sff_trim = list(SffIterator(handle, trim=True))
-
-        from Bio import SeqIO
-        filename = "Roche/E3MFGYR02_random_10_reads_no_trim.fasta"
-        fasta_no_trim = list(SeqIO.parse(filename, "fasta"))
-        filename = "Roche/E3MFGYR02_random_10_reads_no_trim.qual"
-        qual_no_trim = list(SeqIO.parse(filename, "qual"))
-
-        filename = "Roche/E3MFGYR02_random_10_reads.fasta"
-        fasta_trim = list(SeqIO.parse(filename, "fasta"))
-        filename = "Roche/E3MFGYR02_random_10_reads.qual"
-        qual_trim = list(SeqIO.parse(filename, "qual"))
-
-        for s, sT, f, q, fT, qT in zip(sff, sff_trim, fasta_no_trim, qual_no_trim, fasta_trim, qual_trim):
-            assert s.id == f.id == q.id
-            assert str(s.seq) == str(f.seq)
-            assert s.letter_annotations["phred_quality"] == q.letter_annotations["phred_quality"]
-            assert s.id == sT.id == fT.id == qT.id
-            assert str(sT.seq) == str(fT.seq)
-            assert sT.letter_annotations["phred_quality"] == qT.letter_annotations["phred_quality"]
+        with open(filename, "rb") as a_handle, open(filename, "rb") as b_handle:
+            sff = list(SffIterator(a_handle))
+            sff_trim = list(SffIterator(b_handle, trim=True))
+
```

Comment:
At this point (because we used list), both the iterators should have parsed the entire file, and we can close the handle. That was the goal with the original structure:

```
        with open(filename, "rb") as handle:
            sff = list(SffIterator(handle))
        with open(filename, "rb") as handle:
            sff_trim = list(SffIterator(handle, trim=True))
```

Also there is no reason to have the rest of the code indented as part of this with clause.